paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkGenericCatalogFactory.java

```diff
@@ -56,14 +58,24 @@ public Set<org.apache.flink.configuration.ConfigOption<?>> optionalOptions() {
     @Override
     public FlinkGenericCatalog createCatalog(Context context) {
         CatalogFactory hiveFactory = createHiveCatalogFactory(context.getClassLoader());
-        Catalog catalog = hiveFactory.createCatalog(context);
+        Context flinkContext = createFlinkContext(context);
+        Catalog flinkCatalog = hiveFactory.createCatalog(flinkContext);
         return createCatalog(
-                context.getClassLoader(), context.getOptions(), context.getName(), catalog);
+                context.getClassLoader(), context.getOptions(), context.getName(), flinkCatalog);
+    }
+
+    @VisibleForTesting
+    public Context createFlinkContext(Context context) {
+        return new FactoryUtil.DefaultCatalogContext(
+                context.getName(),
+                OptionsUtils.convertToPropertiesPrefixKey(context.getOptions(), HIVE_PREFIX),
```

Comment:
Hive prefix? Why not using `HiveCatalogFactoryOptions`?

Reply:
`HiveCatalogFactoryOptions` is not annotated as `Public`, I think it's better to use `optionalOptions` and `requiredOptions` in `hiveCatalogFactory` to filter specific options in context.

Reply:
@JingsongLi @FangYongs Thx for review. Fixed by using optionalOptions and requiredOptions in CatalogFactory to filter specific options in context.